src/wechaty.ts

```diff
@@ -114,6 +114,15 @@ export interface WechatyOptions {
   ioToken?       : string,                    // Io TOKEN
 }
 
+// A plugin is a Class with a install function.
+export abstract class WechatyPlugin {
```

Comment:
How about a `Functional` interface to archive the same goal?

I'd like to purpose this interface:

```ts
export type WechatyPlugin = (...options: any[]) => (wechaty: Wechaty) => void
```

Reply:
Functional plugin is fine, if you wish to use.
In my project I also use Functional plugin. and I just feel it make code more unreadable.

In webpack, they use Class style plugin. so that's inspired me. 
No need to understand why I have to return function () {};  and what is the `this`. just need to write a plugin class with a  specific method. and you can also write logic in the constructor, or some private method. if the plugin is complex.

but this is fine, if you prefer Functional style, I can change it back.

Reply:
1. No need to understand why I have to return function () {};
1. write the logic in the constructor or some private method. if the plugin is complex.

I believe the above two reasons are quite strong to support the Class style, and I agree with most of your points.

So please feel free to make your final decision for choosing which style we should use. I'll be happy no matter which style you have chosen: `Class` and `Functional` style are all good.